reconcile/ocm_addons_upgrade_tests_trigger.py

```diff
@@ -0,0 +1,92 @@
+import logging
+from typing import Optional
+
+from reconcile import queries
+from reconcile.utils.jenkins_api import JenkinsApi
+from reconcile.utils.ocm import OCMMap
+from reconcile.utils.secret_reader import SecretReader
+from reconcile.utils.state import State
+
+QONTRACT_INTEGRATION = "ocm-addons-upgrade-tests-trigger"
+
+
+def run(dry_run: bool) -> None:
+    settings = queries.get_app_interface_settings()
+    secret_reader = SecretReader(queries.get_secret_reader_settings())
+    accounts = queries.get_state_aws_accounts()
+    state = State(
+        integration=QONTRACT_INTEGRATION, accounts=accounts, settings=settings
+    )
+    ocms = queries.get_openshift_cluster_managers()
+    for ocm_info in ocms:
+        ocm_name = ocm_info["name"]
+        addon_upgrade_tests = ocm_info.get("addonUpgradeTests")
+        if not addon_upgrade_tests:
+            continue
+
+        ocm_map = OCMMap(
+            ocms=[ocm_info],
+            integration=QONTRACT_INTEGRATION,
+            settings=settings,
+            init_addons=True,
+        )
+
+        ocm = ocm_map[ocm_name]
+        state_updates: dict[str, Optional[str]] = {}
+        for aut in addon_upgrade_tests:
+            addon_name = aut["addon"]["name"]
+            addon_org_version = None
+            should_trigger = True
+            state_key = f"{ocm_name}/{addon_name}"
```

Comment:
you may want to take into account @patjlm comment about the state key in https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/55019#note_5540116